### PR TITLE
juliaup: Update to 1.17.2

### PR DESCRIPTION
**Pull request by BoatyPillow**
**Summary**
Updated juliaup to 1.17.2 and resolves getsolus/packages#3111 by updating to the new cargo macros supported by ypkg

Release notes can be found [here](https://github.com/JuliaLang/juliaup/releases/tag/v1.17.2)

**Summary**

**Test Plan**

Installed and tested commands with juliaup using bash
**Checklist**

- [X] Package was built and tested against unstable

### CODE REVIEW

File: packages/j/juliaup/package.yml [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
-    sed -i 's|built::write_built_file().expect("Failed to acquire build-time information");||' build.rs
-    sed -i 's|include!(concat!(env!("OUT_DIR"), "/built.rs"))|pub const PKG_VERSION: \&str = "%version%"|' src/lib.rs
-    cargo fetch --locked
+    %cargo_fetch
 build      : |
-    cargo build --frozen %JOBS% --release --bin juliaup --bin julialauncher --features binjulialauncher
+    %cargo_build --bin juliaup --bin julialauncher --features binjulialauncher
 install    : |
-    install -Dm00755 target/release/{juliaup,julialauncher} -t $installdir/usr/bin
+    %cargo_install
```
Comment (EbonJaeger) [inline packages/j/juliaup/package.yml:21]: This change only installs `juliaup` and not `julialauncher`.

I think you'll have to add another macro call for it.

Comment (Staudey) [reply packages/j/juliaup/package.yml:21]: Should work fine now with the newest `ypkg` (still needs a rebuild of course)

Comment (Staudey) [reply packages/j/juliaup/package.yml:21]: Should work with `%cargo_install juliaup julialauncher` now with the newest `ypkg`

**Comment by davidjharder** [comment]
Hey @BoatyPillow do you have time to look at this?

**Comment by davidjharder** [comment]
Closing in favor of https://github.com/getsolus/packages/pull/4124

**Pull request closed without merging**